src/library_webgpu.js

```diff
@@ -2144,6 +2144,7 @@ var LibraryWebGPU = {
   wgpuRenderPassEncoderSetIndexBuffer: (passId, bufferId, format, offset, size) => {
     var pass = WebGPU.mgrRenderPassEncoder.get(passId);
     var buffer = WebGPU.mgrBuffer.get(bufferId);
+    {{{ gpu.convertSentinelToUndefined('size') }}}
```

Comment:
Works for my use case but, I think you should apply it to the offset, since it is also an optional value that defaults to 0

Reply:
It's not necessary for offset, because in the JS API, 0 and undefined do the same thing - the WebIDL code in the spec also defines the default value as 0. So when C passes 0, we can pass it through directly and it passes through correctly. We only need the special casing for WGPU_WHOLE_SIZE which is a special 0xFFFF_FFFF_FFFF_FFFF which is a placeholder for the JS `undefined`. (it's unsigned in C but when we receive it in JS it's been cast to signed, so we actually receive it as -1, and translate that to `undefined`.)